File: Algorithms/1704. Determine if String Halves Are Alike.rs
impl Solution {
    pub fn halves_are_alike(s: String) -> bool {
        let dic: Vec<char> = vec!['a', 'e',  'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];
        let mid = s.len() / 2;
        let pre =  &s[..mid];
        let post = &s[mid..];
        pre.chars().filter(|x| dic.contains(x)).count() == post.chars().filter(|x| dic.contains(x)).count()
    }
}

File: Algorithms/1704. Determine if String Halves Are Alike.ts
function halvesAreAlike(s: string): boolean {
  let count = 0;
  const vowels = ["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"];
  const chars = s.split("");
  while (chars.length !== 0) {
    const l = chars.shift();
    if (vowels.includes(l)) {
      count += 1;
    }
    const r = chars.pop();
    if (vowels.includes(r)) {
      count -= 1;
    }
  }
  return count === 0;
}
